subprojects/model-core/src/test/groovy/org/gradle/api/internal/provider/CollectionPropertySpec.groovy

```diff
@@ -964,4 +964,61 @@ The value of this property is derived from: <source>""")
     Property<String> elementProperty() {
         return new DefaultProperty<String>(host, String)
     }
+
+    def "runs side effect when calling '#getter' on property to which providers were added via 'add'"() {
+        when:
+        def sideEffect1 = Mock(ValueSupplier.SideEffect)
+        def sideEffect2 = Mock(ValueSupplier.SideEffect)
+        property.add(Providers.of("some value").withSideEffect(sideEffect1))
```

Comment:
Do we need a test here when only one of the two values has a side effect as well?

Reply:
I don't think we need a separate test, but I can add a side-effectless value to this collection as well 👍

Reply:
Done